API.py

```diff
@@ -0,0 +1,87 @@
+import json
+import re
+import uuid
+from flask import Flask, render_template, request, jsonify, abort
+import os
+import time
+
+app = Flask(__name__)
+UPLOAD_FOLDER = 'uploads'
+OUTPUT_FOLDER = 'output'
+app.config['UPLOAD_FOLDER'] = UPLOAD_FOLDER
+app.config['OUTPUT_FOLDER'] = OUTPUT_FOLDER
+the_files = {}
+
+
+@app.route('/')
+def home():
+    return render_template('home.html')
+
+
+@app.route('/upload', methods=['POST'])
+def upload():
+    if 'file' not in request.files:
+        return jsonify({'error': 'No file attached'}), 400
+
+    file = request.files['file']
+    if file.filename == '':
+        return jsonify({'error': 'No file selected'}), 400
+
+    uid = str(uuid.uuid4())
+    timestamp = time.strftime('%d-%m-%Y-%H-%M')
+    original_filename = file.filename.split('.')
+    filename = '.'.join([original_filename[0], timestamp, uid, original_filename[1]])
+    filepath = os.path.join(app.config['UPLOAD_FOLDER'], filename)
+    file.save(filepath)
+    return jsonify({'uid': uid}), 200
+
+
+@app.route('/status/<uid>', methods=['GET'])
+def status(uid):
+    try:
+        output_file_path, file_name = find_file_by_uid(uid, 'OUTPUT_FOLDER')
+    except:
```

Comment:
Always catch the most specific exception.  Read more: https://stackoverflow.com/questions/730764/how-to-properly-ignore-exceptions